diff --git a/node_modules/@module-federation/typescript/dist/src/plugins/FederatedTypesPlugin.js b/node_modules/@module-federation/typescript/dist/src/plugins/FederatedTypesPlugin.js
index 4ffc835..446179e 100644
--- a/node_modules/@module-federation/typescript/dist/src/plugins/FederatedTypesPlugin.js
+++ b/node_modules/@module-federation/typescript/dist/src/plugins/FederatedTypesPlugin.js
@@ -204,7 +204,9 @@ class FederatedTypesPlugin {
                 }
                 if (filesToCacheBust.length > 0) {
                     await Promise.all(filesToCacheBust.filter(Boolean).map((file) => {
-                        const url = new URL(path_1.default.join(origin, typescriptFolderName, file)).toString();
+                        this.logger.log("HEY IT'S ME, I'VE SNUCK IN TO THE NODE MODULES.", { origin, typescriptFolderName, file, pathUrlString: path_1.default.join(origin, typescriptFolderName, file) })
+                        // const url = new URL(path_1.default.join(origin, typescriptFolderName, file)).toString();
+                        const url = new URL(`${origin}/${typescriptFolderName}${file.indexOf('/') === 0 ? '' : '/'}${file}`).toString();
                         const destination = path_1.default.join(this.normalizeOptions.webpackCompilerOptions.context, typescriptFolderName, remote);
                         this.logger.log('Downloading types...');
                         return (0, download_1.default)({
